### PR TITLE
Add codecov configuration

### PR DESCRIPTION
**Pull request by vdemeester**
cc @nandhini915 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    # project will give us the diff in the total code coverage between a commit
+    # and its parent
+    project:
+      default:
+        target: auto
+        threshold: "0.05%"
+    changes: false
+ignore:
+  - "**/internal/test"
+  - "vendor/*"
```
Comment (dnephin) [inline codecov.yml]: Could we ignore generate files too:

```
cli/compose/schema/bindata.go
```

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+comment:
+  layout: header, changes, diff, sunburst
+coverage:
+  status:
+    patch: false
```
Comment (dnephin) [inline codecov.yml]: `patch` coverage seems interesting, could we try it out?

File: codecov.yml (hunk 1 above)
Comment (vdemeester) [inline codecov.yml]: oh right !

Comment (nandhini915) [inline codecov.yml]: Are there any mocks you'd want to ignore? Also integration test modules (though I'm assuming internal/test refers to your system tests)

Comment (vdemeester) [inline codecov.yml]: no mocks for now I think. no integration test either (all are unit tests). `internal/test` is for some utils code (`builders`, …) for the tests, that's why I ignored them 👼

**Comment by nandhini915** [comment]
Also, add https://github.com/docker-codecov-bot to your repo with write access so codecov results can be added to the PR. 
(In addition you could also setup the repo so that the PR can't be merged if coverage drops, but we decided to not add that since it wasn't taking end to end tests into account).

**Comment by dnephin** [comment]
Let's try to get this merged ASAP!

**Pull request merged**